erpnext/stock/get_item_details.py

```diff
@@ -172,10 +172,19 @@ def get_basic_details(args, item):
 		"net_amount": 0.0,
 		"discount_percentage": 0.0,
 		"supplier": item.default_supplier,
+		"update_stock": args.get("update_stock") if args.get('doctype') == 'Sales Invoice' else 0,
 		"delivered_by_supplier": item.delivered_by_supplier if args.get("doctype") in ["Sales Order", "Sales Invoice"] else 0,
 		"is_fixed_asset": item.is_fixed_asset
 	})
 
+	if out.uom != args.uom:
```

Comment:
always set conversion_factor and stock_qty, even if sales uom and stock uom is same.